lib/web_ui/lib/platform_dispatcher.dart

```diff
@@ -492,3 +508,86 @@ enum DartPerformanceMode {
   throughput,
   memory,
 }
+
+abstract class TextScaler {
+  /// Creates a TextScaler.
+  const TextScaler();
+  const factory TextScaler.linear(double textScaleFactor) = _LinearTextScaler;
+
+  static const TextScaler noScaling = _LinearTextScaler(1.0);
+
+  double scale(double fontSize);
+
+  TextScaler clamp(double minScaleFactor, double maxScaleFactor) {
+    return minScaleFactor == maxScaleFactor
+      ? TextScaler.linear(minScaleFactor)
+      : _ClampedTextScaler(this, minScaleFactor, maxScaleFactor);
+  }
+}
+
+final class _LinearTextScaler implements TextScaler {
```

Comment:
Is there a way to share the implementation of this with regular dart:ui? This is just a straight-up copy, right? Would make this easier to maintain...

Reply:
Same with _ClampedTextScaler.

Reply:
(Makes me wonder if dart:ui should maybe just expose the `_getScaledFontSize` and the TestScaler stuff lives in the framework)...